pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -157,6 +157,8 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 	p := cephv1.GetOSDPlacement(c.spec.Placement)
 	if !osdProps.onPVC() {
 		p.ApplyToPodSpec(&podSpec)
+	} else if osdProps.getPreparePlacement().NodeAffinity != nil {
+		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec)
 	} else {
 		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec)
 		p.ApplyToPodSpec(&podSpec)
```

Comment:
If I understand the if-elseif-else block correctly...
- if this is not an OSD on PVC, only apply the OSD placement
- if prepare pod placement node affinity is set, only use prepare pod placement
- if it is an osd on pvc and prepare pod placement isn't set, we still merge prepare and osd spec

This doesn't seem to be quite what is intended.

Also, why do we only care about the `NodeAffinity` property of the prepare placement? We don't care about pod affinity/anti-affinity?

Reply:
going with this (https://github.com/rook/rook/pull/7192#discussion_r573964162) approach